module ExploitDataProxy

  def report_exploit_attempt(host, opts)
    begin
      data_service = self.get_data_service()
      data_service.report_exploit_attempt(host, opts)
    rescue  Exception => e
      self.log_error(e, "Problem reporting exploit attempt")
    end
  end

  def report_exploit_failure(opts)
    begin
      data_service = self.get_data_service()
      data_service.report_exploit_failure(opts)
    rescue  Exception => e
      self.log_error(e, "Problem reporting exploit failure")
    end
  end

  def report_exploit_success(opts)
    begin
      data_service = self.get_data_service()
      data_service.report_exploit_success(opts)
    rescue  Exception => e
      self.log_error(e, "Problem reporting exploit success")
    end
  end
end